indexer/database/contract_events.go

```diff
@@ -218,3 +199,34 @@ func (db *contractEventsDB) L2ContractEventsWithFilter(filter ContractEvent, fro
 
 	return events, nil
 }
+
+// Auxilliary methods for both L1 and L2
+
+// ContractEventsWithFilter will retrieve contract events within the specified range according ot `chain`.
+// NOTE: Not the fondest of the `chain` arg but it unblocks everything else
```

Comment:
Seems fine but if we wanted to do something here easiest thing is to Create l1 and l2 implementations of a struct that have the same interface and start using that instead of naming things l1 and l2 and switching on them.

I think it's fine as is though doing everything in a switch like this is the exact same pattern expressed in a different way that some people prefer

Reply:
Yea it's a bit quirky but was the least-hackiest solution for the `contracts_bedrock/cross_domain_messenger.go` to be cleaner. Almost was gonna explore generics and such but figured I was getting into over-engineering territory when this is a super simple solution.

We can re-think this is it starts popping up on more places. I'm confident though that this is the only place it's needed so that `cross_domain_messenger.go` & `standard_bridge.go` could remain universal and not switch on l1/l2